src/plugins/yawn/listener.cpp

```diff
@@ -0,0 +1,187 @@
+/**
+ * @file
+ *
+ * @brief This file contains the implementation of a basic listener.
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+// -- Imports ------------------------------------------------------------------------------------------------------------------------------
+
+#include <stdexcept>
+
+#include "listener.hpp"
+
+using std::overflow_error;
+using std::string;
+using std::to_string;
+
+using kdb::Key;
+
+// -- Functions ----------------------------------------------------------------------------------------------------------------------------
+
+namespace
+{
+
+/**
+ * @brief This function converts a given number to an array base name.
+ *
+ * @param index This number specifies the index of the array entry.
+ *
+ * @return A string representing the given indices as Elektra array name.
+ */
+string indexToArrayBaseName (uintmax_t const index)
```

Comment:
What about adding elektraArraySetName (Key, size_t) to libease?

Reply:
Sounds like a good idea.